retrofit/src/main/java/retrofit2/Utils.java

```diff
@@ -348,6 +348,19 @@ static Type getParameterUpperBound(int index, ParameterizedType type) {
     return paramType;
   }
 
+  static Type getParameterLowerBound(int index, ParameterizedType type) {
+    Type[] types = type.getActualTypeArguments();
+    if (index < 0 || index >= types.length) {
+      throw new IllegalArgumentException(
```

Comment:
Just let the regular array bounds checking do this?

Reply:
Done